src/@utils/subgraph.ts

```diff
@@ -237,21 +242,17 @@ export async function fetchDataForMultipleChains(
   chainIds: number[]
 ): Promise<any[]> {
   let datas: any[] = []
-  for (const chainId of chainIds) {
-    const context: OperationContext = {
-      url: `${getSubgraphUri(
-        chainId
-      )}/subgraphs/name/oceanprotocol/ocean-subgraph`,
-      requestPolicy: 'network-only'
-    }
-    try {
+  try {
+    for (const chainId of chainIds) {
+      const context: OperationContext = getQueryContext(chainId)
       const response = await fetchData(query, variables, context)
-      datas = datas.concat(response.data)
-    } catch (error) {
-      console.error('Error fetchData: ', error.message)
+      if (!response || response.error) continue
+      datas = datas.concat(response?.data)
     }
+    return datas
+  } catch (error) {
+    LoggerInstance.error('Error fetchData: ', error.message)
```

Comment:
for debugging really helps to be precise in those messages for these generic methods, doesn't help if all of them say they come from `fetchData`. So this here should say `Error fetchDataForMultipleChains`.